rust/lance-encoding/src/compression.rs

```diff
@@ -193,6 +193,36 @@ fn try_bitpack_for_mini_block(_data: &FixedWidthDataBlock) -> Option<Box<dyn Min
     }
 }
 
+fn try_bitpack_for_block(
+    data: &FixedWidthDataBlock,
+) -> Option<(Box<dyn BlockCompressor>, CompressiveEncoding)> {
+    let bits = data.bits_per_value;
+    if !matches!(bits, 8 | 16 | 32 | 64) {
+        return None;
+    }
+
+    let bit_widths = data.expect_stat(Stat::BitWidth);
+    let widths = bit_widths.as_primitive::<UInt64Type>();
+    let has_all_zeros = widths.values().contains(&0);
+    let too_small =
+        widths.len() == 1 && InlineBitpacking::min_size_bytes(widths.value(0)) >= data.data_size();
+
+    if has_all_zeros || too_small {
+        return None;
+    }
+
+    if data.num_values <= 1024 {
```

Comment:
Why not just use out of line bitpacking for both?

Reply:
Ah, I see now.  Inline bitpacking currently can't handle more than one chunk when decompressing.  Out of line bitpacking is not very efficient at the moment when handling a single chunk.